### DISCUSSION TITLE
How to add typst-lsp to Eglot ?

### DISCUSSION THREAD

**Discussion by raylemon** [abougouffa/minemacs#153] [General]
I want to add to eglot typst-lsp. I already have set typst-ts-mode to tree-sitter, but when I write
```elisp
(+eglot-register
'(typst-ts-mode) "typst-lsp")
```
It seems not recognized.

**Comment by abougouffa** [comment]
Can you try `(+eglot-register 'typst-ts-mode "typst-lsp")` please?

**Reply by abougouffa** [reply]
Can you also check if when you open a typst file, the `typst-ts-mode` is the one which is running. Otherwise, you can try to add both TS and classic modes like this:

```elisp
(+eglot-register '(typst-ts-mode typst-mode) "typst-lsp")
```
